From dfee4e1130556b3e09ccabdd32498cef8b4ff614 Mon Sep 17 00:00:00 2001
From: Josue David Hernandez <josue.d.hernandez.gutierrez@intel.com>
Date: Wed, 2 May 2018 15:41:26 -0500
Subject: [PATCH] clr-boot-manager: set systemd initrd options

our systemd versions don't take rd.luks.options=timeout=0 (default)
as unlimited, it means there is not time to provide the password for an
encrypted device

the combination of rd.luks.uuid and root=UUID= not work in old systemd versions, this patch force CBM to use
rd.luks.name=<UUID>=clr-root instead, it means root will point to /dev/mapper/clr-root

Signed-off-by: Josue David Hernandez <josue.d.hernandez.gutierrez@intel.com>
---
 src/bootloaders/systemd-class.c | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/src/bootloaders/systemd-class.c b/src/bootloaders/systemd-class.c
index b09497f..22658fa 100644
--- a/src/bootloaders/systemd-class.c
+++ b/src/bootloaders/systemd-class.c
@@ -260,16 +260,20 @@ bool sd_class_install_kernel(const BootManager *manager, const Kernel *kernel)
         }
 
         /* Add the root= section */
-        if (root_dev->part_uuid) {
-                cbm_writer_append_printf(writer, "options root=PARTUUID=%s ", root_dev->part_uuid);
-        } else {
-                cbm_writer_append_printf(writer, "options root=UUID=%s ", root_dev->uuid);
-        }
-        /* Add LUKS information if relevant */
         if (root_dev->luks_uuid) {
-                cbm_writer_append_printf(writer, "rd.luks.uuid=%s ", root_dev->luks_uuid);
+                /* Add LUKS information if relevant */
+                cbm_writer_append_printf(writer, "options root=/dev/mapper/clr-root ");
+                cbm_writer_append_printf(writer, "rd.luks.name=%s=clr-root ", root_dev->luks_uuid);
+                cbm_writer_append_printf(writer, "rd.luks.options=timeout=90 ");
+        } else {
+                if (root_dev->part_uuid) {
+                        cbm_writer_append_printf(writer, "options root=PARTUUID=%s ", root_dev->part_uuid);
+                } else {
+                        cbm_writer_append_printf(writer, "options root=UUID=%s ", root_dev->uuid);
+                }
         }
 
+
         /* Finish it off with the command line options */
         cbm_writer_append_printf(writer, "%s\n", kernel->meta.cmdline);
         cbm_writer_close(writer);
-- 
2.17.0

